keys.go

```diff
@@ -161,32 +160,27 @@ func (privk *PrivateKey) WriteToFile(filename string, forceOverwrite bool) (int6
 	return privk.WriteTo(f)
 }
 
-func (privk *PrivateKey) RevocationKey() (*revocation.PrivateKey, error) {
-	if privk.revocationKey == nil {
-		if !privk.RevocationSupported() {
-			return nil, errors.New("private key does not support revocation")
-		}
-		bts, err := base64.StdEncoding.DecodeString(privk.ECDSA)
-		if err != nil {
-			return nil, err
-		}
-		key, err := signed.UnmarshalPrivateKey(bts)
-		if err != nil {
-			return nil, err
-		}
-		privk.revocationKey = &revocation.PrivateKey{
-			Counter: privk.Counter,
-			ECDSA:   key,
-			P:       privk.PPrime,
-			Q:       privk.QPrime,
-			N:       new(big.Int).Mul(privk.P, privk.Q),
-		}
+func (privk *PrivateKey) parseRevocationKey() error {
```

Comment:
`ECDSA` can't be made unexported, because it is used in another subpackage (`revocation`). We could, however, rename it to `RevocationKey`.

However, this name sort of excludes the future possibility that we might use this key for other purposes than just revocation (although I don't currently have any such use cases in mind). In addition, the XML serialized form of private and public keys already contain XML tags called `<ECDSA>` ([example](https://github.com/privacybydesign/pbdf-schememanager/blob/master/ivido/PublicKeys/1.xml#L29)). So we can rename this member here, but then we either have to accept that we create an inconsistency between this name and the name of the corresponding XML tag, or we additionally have to create some migration path for the XML tag name.

All in all, I'd say keeping the current name is easiest, if not ideal.